code/game/machinery/computer/mechlaunchpad.dm

```diff
@@ -0,0 +1,150 @@
+	else
+		return ..()
+
+/**
+  * Tries to call the launch proc on the connected mechpad, returns if there is no connected mechpad or there is no mecha on the pad
+  * Arguments:
+  * * user - The user of the proc
+  * * where - The mechpad that the connected mechpad will try to send a supply pod to
+  */
+/obj/machinery/computer/mechpad/proc/try_launch(var/mob/user, var/obj/machinery/mechpad/where)
+	if(!connected_mechpad)
+		to_chat(user, "<span class='warning'>[src] has no connected pad!</span>")
+		return
+	if(locate(/obj/mecha) in get_turf(connected_mechpad))
+		connected_mechpad.launch(where)
+	else
+		to_chat(user, "<span class='warning'>[src] detects no mecha on the pad!</span>")
+
+///Checks if the pad of a certain number has been QDELETED, if yes returns FALSE, otherwise returns TRUE
+/obj/machinery/computer/mechpad/proc/pad_exists(number)
+	var/obj/machinery/mechpad/pad = mechpads[number]
+	if(QDELETED(pad))
+		return FALSE
+	return TRUE
+
+///Returns the pad of the value specified
+/obj/machinery/computer/mechpad/proc/get_pad(number)
+	var/obj/machinery/mechpad/pad = mechpads[number]
+	return pad
+
+/obj/machinery/computer/mechpad/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "MechpadConsole", name, ui_x, ui_y, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/mechpad/ui_data(mob/user)
+	var/list/data = list()
+	var/list/pad_list = list()
+	for(var/i in 1 to LAZYLEN(mechpads))
+		if(pad_exists(i))
+			var/obj/machinery/mechpad/pad = get_pad(i)
+			var/list/this_pad = list()
+			this_pad["name"] = pad.display_name
+			this_pad["id"] = i
+			if(pad.machine_stat & NOPOWER)
+				this_pad["inactive"] = TRUE
+			pad_list += list(this_pad)
+		else
+			mechpads -= get_pad(i)
+	data["mechpads"] = pad_list
+	data["selected_id"] = selected_id
+	data["connected_mechpad"] = !!connected_mechpad
+	if(selected_id)
+		var/obj/machinery/mechpad/current_pad = mechpads[selected_id]
+		data["pad_name"] = current_pad.display_name
+		data["selected_pad"] = current_pad
+		if(QDELETED(current_pad) || (current_pad.machine_stat & NOPOWER))
+			data["pad_active"] = FALSE
+			return data
+		data["pad_active"] = TRUE
+	return data
+
+/obj/machinery/computer/mechpad/ui_act(action, params)
+	if(..())
+		return
+	var/obj/machinery/mechpad/current_pad = mechpads[selected_id]
+	switch(action)
+		if("select_pad")
+			selected_id = text2num(params["id"])
+			. = TRUE
+		if("rename")
+			. = TRUE
+			var/new_name = params["name"]
+			if(!new_name)
+				return
+			current_pad.display_name = new_name
+		if("remove")
+			if(usr && alert(usr, "Are you sure?", "Unlink Orbital Pad", "I'm Sure", "Abort") != "Abort")
+				mechpads -= current_pad
+				selected_id = null
+			. = TRUE
+		if("launch")
+			try_launch(usr, current_pad)
+			. = TRUE
+	. = TRUE
```

Comment:
Judging by the existing cases you could just add `. = TRUE` before the `switch()` and remove the rest